pkg/chrootarchive/archive.go

```diff
@@ -48,16 +46,20 @@ func untarHandler(tarArchive io.Reader, dest string, options *archive.TarOptions
 		}
 	}
 
-	r := ioutil.NopCloser(tarArchive)
+	r := tarArchive
```

Comment:
> The http package does this and it causes havoc trying to figure out why things are getting closed at inopportune times.

I am not sure what's that about, if you have some reference; maybe that's something big needs golang.org team to fix?